apps/aecore/test/aecore_suite_utils.erl

```diff
@@ -324,21 +308,27 @@ mine_blocks_until_txs_on_chain(Node, TxHashes, MiningRate, Max) ->
 mine_blocks_until_txs_on_chain(Node, TxHashes, MiningRate, Max, Opts) ->
     %% Fail early rather than having to wait until max_reached if txs already on-chain
     ok = assert_not_already_on_chain(Node, TxHashes),
-    ok = rpc:call(Node, application, set_env,
-                  [aecore, expected_mine_rate, MiningRate], 5000),
-    [] = flush_new_blocks(),
-    subscribe(Node, block_created),
-    StartRes = rpc:call(Node, aec_conductor, start_mining, [Opts], 5000),
-    ct:log("aec_conductor:start_mining() (~p) -> ~p", [Node, StartRes]),
-    Res = mine_blocks_until_txs_on_chain_loop(Node, TxHashes, Max, []),
-    StopRes = rpc:call(Node, aec_conductor, stop_mining, [], 5000),
-    ct:log("aec_conductor:stop_mining() (~p) -> ~p", [Node, StopRes]),
-    unsubscribe(Node, block_created),
-    case Res of
-        {ok, BlocksReverse} ->
-            {ok, lists:reverse(BlocksReverse)};
-        {error, Reason} ->
-            erlang:error(Reason)
+    Fun = fun() ->
+              mine_blocks_until_txs_on_chain_loop(Node, TxHashes, Max, [])
+          end,
+    mine_safe_setup(Node, MiningRate, Opts, Fun).
+
+mine_micro_block_emptying_mempool_or_fail(Node) ->
+    Fun = fun() ->
+              mine_blocks_loop(2, any)
+          end,
+    {ok, Blocks} = mine_safe_setup(Node, expected_mine_rate(), #{}, Fun),
+    [key, micro] = [aec_blocks:type(B) || B <- Blocks],
+
+    case rpc:call(Node, aec_tx_pool, peek, [infinity]) of
+        {ok, []} ->
+            {ok, Blocks};
+        {ok, [_|_]} ->
+            %% So, Tx(s) is/are back in the mempool, this means (unless some Txs arrived
+            %% from thin air) that we had a micro-fork. Let's check what state we stopped in
+            {ok, NewBlocks} = mine_safe_setup(Node, expected_mine_rate(), #{}, Fun),
+            [key, micro] = [aec_blocks:type(B) || B <- NewBlocks],
```

Comment:
Why is `NewBlocks` expected to be a list of two blocks, one key block followed by a microblock?

Reply:
I just used the assertion from the previous setup which expects the transactions in the mempool to appear in a microblock next. If any other situation occurs the test will crash here which is what is expected.